bundles/org.openhab.binding.haywardomnilogic/src/main/java/org/openhab/binding/haywardomnilogic/internal/HaywardThingHandler.java

```diff
@@ -0,0 +1,122 @@
+ * Class to each Hayward Thing Handler
+ *
+ * @author Matt Myers - Initial contribution
+ */
+
+@NonNullByDefault
+public abstract class HaywardThingHandler extends BaseThingHandler {
+
+    public HaywardThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    public abstract void getTelemetry(String xmlResponse) throws HaywardException;
+
+    public State toState(String type, String channelID, String value) throws NumberFormatException {
+        switch (type) {
+            case "Number":
+                return new DecimalType(value);
+            case "Switch":
+                return Integer.parseInt(value) > 0 ? OnOffType.ON : OnOffType.OFF;
+            case "system.power":
+                return Integer.parseInt(value) > 0 ? OnOffType.ON : OnOffType.OFF;
+            case "Number:Dimensionless":
+                switch (channelID) {
+                    case "chlorTimedPercent":
+                    case "filterSpeed":
+                    case "pumpSpeed":
+                    case "filterLastSpeed":
+                        return new QuantityType<>(Integer.parseInt(value), Units.PERCENT);
+                    case "chlorAvgSaltLevel":
+                    case "chlorInstantSaltLevel":
+                        return new QuantityType<>(Integer.parseInt(value), Units.PARTS_PER_MILLION);
+                }
+            case "Number:Temperature":
+                Bridge bridge = getBridge();
+                if (bridge != null) {
+                    HaywardBridgeHandler bridgehandler = (HaywardBridgeHandler) bridge.getHandler();
+                    if (bridgehandler != null) {
+                        if (bridgehandler.account.units.equals("Standard")) {
+                            return new QuantityType<>(Integer.parseInt(value), ImperialUnits.FAHRENHEIT);
+                        } else {
+                            return new QuantityType<>(Integer.parseInt(value), SIUnits.CELSIUS);
+                        }
+                    } else {
+                        // default to imperial if no bridge
+                        return new QuantityType<>(Integer.parseInt(value), ImperialUnits.FAHRENHEIT);
+                    }
+                }
+            default:
+                return StringType.valueOf(value);
+        }
+    }
+
+    public String cmdToString(Command command) {
+        if (command == OnOffType.OFF) {
+            return "0";
+        } else if (command == OnOffType.ON) {
+            return "1";
+        } else if (command instanceof DecimalType) {
+            return ((DecimalType) command).toString();
+        } else if (command instanceof QuantityType) {
+            return ((QuantityType<?>) command).format("%1.0f");
+        } else {
+            return ((StringType) command).toString();
```

Comment:
You don't need to cast to call `toString()`.
```suggestion
            return command.toString();
```